### ISSUE TITLE
id-blacklist doesn't warn when renaming properties to blacklisted identifiers in destructuring assignment.

### ISSUE THREAD

**Issue by jpramassini** [eslint/eslint#12807]
**Tell us about your environment**

* **ESLint Version:** 7.0.0 (upcoming)
* **Node Version:** 10.15.3
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** 
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
id-blacklist: ["foo", "bar"] // Rule options
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const {foo: bar} = baz
```

**What did you expect to happen?**
ESLint should warn that `bar` is a blacklisted identifier, but not `foo` as it is a property from the original `baz` object.

**What actually happened? Please include the actual, raw output from ESLint.**
There is no warning for either identifier.

**Context for this:** A recent PR bugfix removed the errant warning on properties that were being renamed (`foo` in the example above). In the course of this it ignores the following blacklisted `bar` identifier. This is a follow up to https://github.com/eslint/eslint/pull/12792.

**Are you willing to submit a pull request to fix this bug?**
Yes

**Comment by mdjermanovic** [comment]
Thanks!

If I understand correctly, this is a regression introduced in #12792.

**Comment by jpramassini** [comment]
That's correct. After working with @kaicataldo on it, he suggested that these two issues with the rule be resolved in separate issues/PRs.

**Comment by mdjermanovic** [comment]
Thanks for the follow-up issue! Marked as accepted.

In addition to the original example, I think the following should be also a warning (when `foo` is blacklisted):

```js
const { foo } = baz;
```

**Comment by mdjermanovic** [comment]
Just to check as it would be nice to fix this for `v7.0.0` - is anyone working on this? :)

**Comment by kaicataldo** [comment]
I don't think so!

**Comment by yeonjuan** [comment]
If there is no one working on, I will try. : )
